packages/compass-connections/src/stores/connections-store-redux.ts

```diff
@@ -1479,60 +1479,67 @@ const connectWithOptions = (
       return inflightConnection;
     }
     inflightConnection = (async () => {
-      const isAutoconnectAttempt = isAutoconnectInfo(
-        getState(),
-        connectionInfo.id
-      );
-
       const deviceAuthAbortController = new AbortController();
 
-      connectionInfo = cloneDeep(connectionInfo);
+      try {
```

Comment:
Fun bug I found in connect logic while working on this: because we only clean up `inflightConnection` in the `finally`, if you short circuit before that, you can never reconnect 🙈 Before that this was only affecting `maximumNumberOfActiveConnections` and I'm just assuming that it was a corner-case enough that nobody ever ran into this yet